src/Native/Windows/CMakeLists.txt

```diff
@@ -1,37 +1,132 @@
+SET (CMAKE_INCLUDE_CURRENT_DIR              ON)
+SET (CMAKE_SHARED_LIBRARY_PREFIX            "")
+  
+# Force an out of source build
 if("${CMAKE_SOURCE_DIR}" STREQUAL "${CMAKE_BINARY_DIR}")
     message(FATAL_ERROR "Binary directory isn't being correctly set before calling Cmake. Tree must be built in separate directory from source.")
 endif()
 
 project(CoreFX)
-  
-set(CMAKE_INSTALL_PREFIX $ENV{__CMakeBinDir})
-set(CMAKE_INCLUDE_CURRENT_DIR ON)
-set(CMAKE_CXX_FLAGS "-std=c++11")
-set(CMAKE_SHARED_LIBRARY_PREFIX "")
+string(TOUPPER $ENV{CMAKE_BUILD_TYPE} UPPERCASE_CMAKE_BUILD_TYPE)
 
-if (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 3.5)
-    add_compile_options(-Wno-unreachable-code)
+# The following options are set by the razzle build
+add_compile_options(/d2Zi+)       # make optimized builds debugging easier
+add_compile_options(/nologo)      # Suppress Startup Banner
+add_compile_options(/Oi)          # enable intrinsics
+add_compile_options(/Oy-)         # disable suppressing of the creation of frame pointers on the call stack for quicker function calls
+add_compile_options(/U_MT)        # undefine the predefined _MT macro
+add_compile_options(/GF)          # enable read-only string pooling
+add_compile_options(/Gm-)         # disable minimal rebuild
+add_compile_options(/EHa)         # enable C++ EH (w/ SEH exceptions)
+add_compile_options(/Zp8)         # pack structs on 8-byte boundary
+add_compile_options(/Gy)          # separate functions for linker
+add_compile_options(/Zc:wchar_t-) # C++ language conformance: wchar_t is NOT the native type, but a typedef
+add_compile_options(/Zc:forScope) # C++ language conformance: enforce Standard C++ for scoping rules
+add_compile_options(/GR-)         # disable C++ RTTI
+add_compile_options(/FC)          # use full pathnames in diagnostics
+add_compile_options(/MP)          # Build with Multiple Processes (number of processes equal to the number of processors)
+add_compile_options(/GS)          # Buffer Security Check
+add_compile_options(/Zm200)       # Specify Precompiled Header Memory Allocation Limit of 150MB
+add_compile_options(/Zi)          # enable debugging information
+add_compile_options(/Zl)          # enable debugging information
+add_compile_options(/wd4960 /wd4961 /wd4603 /wd4627 /wd4838 /wd4456 /wd4457 /wd4458 /wd4459 /wd4091 /we4640)
+
+if ($ENV{__BuildArch} STREQUAL "x86")
+    add_compile_options(/Gz)
+endif ()
+if(NOT $ENV{__BuildArch} STREQUAL "arm64")
+    # enable control-flow-guard support for native components for non-Arm64 builds
+    add_compile_options(/guard:cf) 
+    set(__LinkArgs "${__LinkArgs} /guard:cf")
 endif ()
 
-if (CMAKE_SYSTEM_PROCESSOR STREQUAL x86_64 OR CMAKE_SYSTEM_PROCESSOR STREQUAL amd64)
+SET (CMAKE_C_STANDARD_LIBRARIES "") # do not link against standard win32 libs i.e. kernel32, uuid, user32, etc.
+SET (CMAKE_CXX_STANDARD_LIBRARIES "") # do not link against standard win32 libs i.e. kernel32, uuid, user32, etc.
```

Comment:
These don't actually do anything, interestingly enough. The win32 libs will be linked regardless of the libraries passed to the linker. I think adding the /MT(d) argument may cause them to be linked behind the scenes.

Either way, the only thing these lines are <i>actually</i> accomplishing is removing "win32.lib" etc. from the linker arguments. The resulting binary still imports them.

I meant to remove these. Pushing an update to that now.